ContosoUniversity/Program.cs

```diff
@@ -1,22 +1,70 @@
-﻿using Microsoft.AspNetCore.Hosting;
+﻿using ContosoUniversity.Data;
+using ContosoUniversity.Infrastructure;
+using ContosoUniversity.Infrastructure.Tags;
+using FluentValidation.AspNetCore;
+using HtmlTags;
+using MediatR;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
 
-namespace ContosoUniversity
+var builder = WebApplication.CreateBuilder(args);
+
+RegisterServices(builder);
+
+var app = builder.Build();
+
+ConfigureApplication(app);
+
+app.Run();
+
+static void RegisterServices(WebApplicationBuilder builder)
 {
-    public class Program
-    {
-        public static void Main(string[] args)
+    var services = builder.Services;
+
+    services.AddMiniProfiler().AddEntityFramework();
+
+    services.AddDbContext<SchoolContext>(options =>
+        options.UseSqlServer(builder.Configuration.GetConnectionString("DefaultConnection")));
+
+    services.AddAutoMapper(typeof(Program));
+
+    services.AddMediatR(typeof(Program));
+
+    services.AddHtmlTags(new TagConventions());
+
+    services.AddRazorPages(opt =>
         {
-            var host = CreateHostBuilder(args).Build();
+            opt.Conventions.ConfigureFilter(new DbContextTransactionPageFilter());
+            opt.Conventions.ConfigureFilter(new ValidatorPageFilter());
+        })
+        .AddFluentValidation(cfg => { cfg.RegisterValidatorsFromAssemblyContaining<Program>(); });
 
-            host.Run();
-        }
+    services.AddMvc(opt => opt.ModelBinderProviders.Insert(0, new EntityModelBinderProvider()));
+}
+
+static void ConfigureApplication(WebApplication app)
+{
+    app.UseMiniProfiler();
 
-        public static IHostBuilder CreateHostBuilder(string[] args) =>
-            Host.CreateDefaultBuilder(args)
-                .ConfigureWebHostDefaults(webBuilder =>
-                {
-                    webBuilder.UseStartup<Startup>();
-                });
+    if (!app.Environment.IsDevelopment())
+    {
+        app.UseExceptionHandler("/Error");
+        app.UseHsts();
     }
-}
+    else
+    {
+        app.UseDeveloperExceptionPage();
```

Comment:
According to the [docs](https://docs.microsoft.com/en-us/aspnet/core/migration/50-to-60](https://docs.microsoft.com/en-us/aspnet/core/migration/50-to-60#differences-between-the-aspnet-core-5-and-6-hosting-models)). It is no longer required:
> In development mode, the developer exception page middleware is enabled by default.

Reply:
Ah thanks! I created a new PR to incorporate this